tests/datasets/test_transfer.py

```diff
@@ -208,6 +222,37 @@ def second_chunk_request_matcher(request):
     )
 
 
+@pytest.mark.parametrize("max_chunks, expected_chunks", [(10, 10), (100, 20)])
+def test_s3_dynamic_chunk_upload(
+    mocker, mock_client_upload_s3, requests_mock, max_chunks, expected_chunks
+):
+    mocker.patch("faculty.datasets.transfer.DEFAULT_CHUNK_SIZE", 100)
+    mocker.patch("faculty.datasets.transfer.S3_MAX_CHUNKS", max_chunks)
+    chunk_size = transfer._s3_chunk_size(len(TEST_CONTENT))
+    chunks = transfer._rechunk_data([TEST_CONTENT], chunk_size)
+    chunk_matchers = [
+        lambda x: next(chunks) == x.text.encode("utf-8")
+        for i in range(max_chunks)
+    ]
+    urls = [
+        "https://example.com/presigned-url-{i}/url".format(i=i)
+        for i in range(max_chunks)
+    ]
+    tags = ["tag-{tagid}".format(tagid=i) for i in range(max_chunks)]
+
+    for i, (url, cm, tag) in enumerate(zip(urls, chunk_matchers, tags)):
```

Comment:
You're not using the `i` generated by `enumerate` - remove it?

Could we also rename tag -> etag and cm -> matcher (or similar), for explicitness?